### ISSUE TITLE
"ns "java.lang.StringBuilder" not available {:ns java.lang.StringBuilder}"

### ISSUE THREAD

**Issue by ocharles** [thheller/shadow-build#22]
Hi thheller, afraid another bug has come along now! Oddly running a build in my normal development environment works fine, but when I run it on the continuous integration build machine, I get the following:

```
Resolving deps for: inflections.core
Resolving deps for: no.en.core
Resolving deps for: enfocus.bind
Resolving deps for: fynder.winchan
Resolving deps for: fynder.devel
Exception in thread "main" clojure.lang.ExceptionInfo: ns "java.lang.StringBuilder" not available {:ns java.lang.StringBuilder}, compiling:(/tmp/form-init176839270867670870.clj:1:72)
	at clojure.lang.Compiler.load(Compiler.java:7249)
	at clojure.lang.Compiler.loadFile(Compiler.java:7175)
	at clojure.main$load_script.invoke(main.clj:275)
	at clojure.main$init_opt.invoke(main.clj:280)
	at clojure.main$initialize.invoke(main.clj:308)
	at clojure.main$null_opt.invoke(main.clj:343)
	at clojure.main$main.doInvoke(main.clj:421)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at clojure.lang.Var.invoke(Var.java:383)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.Var.applyTo(Var.java:700)
	at clojure.main.main(main.java:37)
Caused by: clojure.lang.ExceptionInfo: ns "java.lang.StringBuilder" not available {:ns java.lang.StringBuilder}
	at clojure.core$ex_info.invoke(core.clj:4591)
	at shadow.cljs.build$get_deps_for_ns_STAR_.invoke(build.clj:417)
	at clojure.core.protocols$iter_reduce.invoke(protocols.clj:49)
	at clojure.core.protocols$fn__6504.invoke(protocols.clj:92)
	at clojure.core.protocols$fn__6448$G__6443__6461.invoke(protocols.clj:13)
	at clojure.core$reduce.invoke(core.clj:6515)
	at shadow.cljs.build$get_deps_for_ns_STAR_.invoke(build.clj:426)
	at clojure.core.protocols$iter_reduce.invoke(protocols.clj:49)
	at clojure.core.protocols$fn__6504.invoke(protocols.clj:92)
	at clojure.core.protocols$fn__6448$G__6443__6461.invoke(protocols.clj:13)
	at clojure.core$reduce.invoke(core.clj:6515)
	at shadow.cljs.build$get_deps_for_ns_STAR_.invoke(build.clj:426)
	at clojure.core.protocols$iter_reduce.invoke(protocols.clj:49)
	at clojure.core.protocols$fn__6504.invoke(protocols.clj:92)
	at clojure.core.protocols$fn__6448$G__6443__6461.invoke(protocols.clj:13)
	at clojure.core$reduce.invoke(core.clj:6515)
	at shadow.cljs.build$get_deps_for_ns_STAR_.invoke(build.clj:426)
	at clojure.core.protocols$iter_reduce.invoke(protocols.clj:49)
	at clojure.core.protocols$fn__6504.invoke(protocols.clj:92)
	at clojure.core.protocols$fn__6448$G__6443__6461.invoke(protocols.clj:13)
	at clojure.core$reduce.invoke(core.clj:6515)
	at shadow.cljs.build$get_deps_for_ns_STAR_.invoke(build.clj:426)
	at shadow.cljs.build$get_deps_for_ns.invoke(build.clj:438)
	at shadow.cljs.build$resolve_main_deps.invoke(build.clj:824)
	at clojure.lang.PersistentVector.reduce(PersistentVector.java:333)
	at clojure.core$reduce.invoke(core.clj:6514)
	at shadow.cljs.build$do_analyze_module.invoke(build.clj:1017)
	at clojure.core.protocols$iter_reduce.invoke(protocols.clj:49)
	at clojure.core.protocols$fn__6508.invoke(protocols.clj:102)
	at clojure.core.protocols$fn__6448$G__6443__6461.invoke(protocols.clj:13)
	at clojure.core$reduce.invoke(core.clj:6515)
	at shadow.cljs.build$step_compile_modules$fn__6393.invoke(build.clj:1072)
	at shadow.cljs.build$step_compile_modules.invoke(build.clj:1069)
	at fynder.shadowbuild$prod.doInvoke(shadowbuild.clj:60)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at clojure.lang.Var.invoke(Var.java:375)
	at user$eval5.invoke(form-init176839270867670870.clj:1)
	at clojure.lang.Compiler.eval(Compiler.java:6792)
	at clojure.lang.Compiler.eval(Compiler.java:6782)
	at clojure.lang.Compiler.load(Compiler.java:7237)
	... 11 more
```

`project.clj` is:

```clojure
(defproject fynder-mobile "0.1.0"
  :dependencies [
                 ;; Clojure
                 [org.clojure/clojure "1.7.0-beta2"]
                 [io.fynder/core.async "0.1.0-SNAPSHOT"]
                 [org.clojure/core.match "0.3.0-alpha4"]
                 [environ "1.0.0"]
                 [compojure "1.3.2" :exclusions [ring/ring-core]]
                 [ring "1.3.2"]
                 [prismatic/schema "0.3.7"]
                 [enlive "1.1.5"]
                 [optimus "0.17.0"]
                 [optimus-less "0.2.1"]
                 [stasis "2.2.2" :exclusions [org.clojure/clojure]]
                 [clj-webdriver "0.6.1"]
                 [org.clojure/core.cache "0.6.4"]

                 ;; ClojureScript
                 [org.clojure/clojurescript "0.0-3269"]
                 [cljsjs/react-with-addons "0.12.2-7"]
                 [thheller/shadow-build "1.0.0-alpha11"]
                 [noencore "0.1.16"]
                 [org.omcljs/om "0.8.8" :exclusions [cljsjs/react]]
                 [sablono "0.3.4"]
                 [enfocus "2.1.0"]
                 [com.cemerick/url "0.1.1"]]
  :source-paths ["src/clj" "src/cljs"]
  :test-paths ["test/clj" "test/cljs"]
  :test-selectors {:default (complement :acceptance)
                   :acceptance :acceptance}
  :resource-paths ["resources"]
  :plugins [[lein-environ "0.5.0"]
            [lein-localrepo "0.5.3"]
            [com.cemerick/clojurescript.test "0.3.1"]
            [hiccup-bridge "1.0.1"]]
  :aliases {"cljs" ["run" "-m" "fynder.shadowbuild/dev"]
            "web" ["run" "-m" "fynder.webserver/web"]
            "acceptance" ["test" ":acceptance"]
            "export" ["run" "-m" "fynder.exporter"]
            "publish" ["with-profile" "prod" ["do"
                                              ["run" "-m" "fynder.shadowbuild/loader"]
                                              ["run" "-m" "fynder.shadowbuild/prod"]
                                              ["export"]]]}
  :local-repo ~(System/getenv "LOCAL_REPO") ; The deps repository can be overridden by a LOCAL_REPO variable.
  :repositories [["sonatype" {:url "http://oss.sonatype.org/content/repositories/releases"}]]
  :profiles {:dev {:repl-options {:init-ns fynder.webserver :timeout 120000}
                   :env {:logging-enabled? true
                         :analytics-enabled? false
                         :api-server "https://m-staging.fynder.io:443"}}})
```

Also, `fynder.devel` is:

```clojure
(ns fynder.devel
  (:require [clojure.browser.repl :as repl]))
```

(nothing more)

I can only conclude it's due to a discrepancy in my environment variables, so I'm going to investigate that.

**Comment by ocharles** [comment]
My guess was wrong, removing `(cljs/step-configure-module :devel '[fynder.devel] #{:cljs})` from `define-modules` in `shadowbuild.clj` has fixed it.

**Comment by thheller** [comment]
That is a pretty weird error since ```java.lang.StringBuilder``` should not be in any ```.cljs``` file.

**Comment by thheller** [comment]
Only thing I can think of is a ```.cljc``` files sitting somewhere that does ```(:import java.lang.StringBuilder)``` or ```:require```?

**Comment by ocharles** [comment]
I've grepped my entire source tree for `StringBuilder` and there are no hits - so if there is a cljc file it's not in my code...

**Comment by thheller** [comment]
Can you put this into your ```shadowbuild.clj``` and call the function just before ```step-compile-modules```? 

```clojure
(defn find-the-villain [state]
  (doseq [{:keys [name requires] :as rc} (-> state :sources vals)
          :when (contains? requires 'java.lang.StringBuilder)]
    (prn [:villain name rc]))
  state)
```

**Comment by thheller** [comment]
```clojure
(-> state
    (find-the-villain)
    (step-compile-modules)
    ...)
```

**Comment by thheller** [comment]
I think I figured it out. It is probably caused by a conflict in cljs where a ```src/clj/cljs/repl.cljc``` and ```src/main/cljs/cljs/repl.cljs``` exist and both try to provide ```cljs.repl```. Seems like shadow-build might pick the wrong one under certain circumstances leaking JVM specific structure into the CLJS build. Could not reproduce it yet but opened a JIRA issue for CLJS, hopefully they will revert the change.

http://dev.clojure.org/jira/browse/CLJS-1289

**Comment by thheller** [comment]
I just committed a fix for the ```cljs.repl``` situation, should fix your issues.

Released ```[thheller/shadow-build "1.0.0-alpha12"]```.
